diff --git a/node_modules/@koinos/sdk-as/assembly/util/mockVM.ts b/node_modules/@koinos/sdk-as/assembly/util/mockVM.ts
index d784e23..204d9b3 100644
--- a/node_modules/@koinos/sdk-as/assembly/util/mockVM.ts
+++ b/node_modules/@koinos/sdk-as/assembly/util/mockVM.ts
@@ -293,6 +293,21 @@ export namespace MockVM {
     System.putObject(METADATA_SPACE, 'verify_vrf', verifyVRFProofResultListType, value.list_type.encode);
   }
 
+  export function getCallContractArguments(): system_calls.call_arguments[] {
+    const callContractArgumentsListType = System.getObject<string, value.list_type>(METADATA_SPACE, "call_contract_arguments", value.list_type.decode);
+    const callContractArguments = new Array<system_calls.call_arguments>(
+      callContractArgumentsListType!.values.length
+    );
+    for (let i = 0; i < callContractArgumentsListType!.values.length; i += 1) {
+      callContractArguments[i] = Protobuf.decode<system_calls.call_arguments>(callContractArgumentsListType!.values[i].bytes_value, system_calls.call_arguments.decode);
+    }
+    return callContractArguments;
+  }
+
+  export function clearCallContractArguments(): void {
+    System.removeObject(METADATA_SPACE, "call_contract_arguments");
+  }
+
   /**
     * Set call contract results that will be used when calling System.callContract(...)
     * @param { Uint8Array[] } callContractResults The results are FIFO, so the first System.callContract(...) used in your code will use the first result you set in callContractResults, the second System.callContract(...) will get the second result, etc...
